src/redux/books/books.js

```diff
@@ -0,0 +1,51 @@
+import { createSlice, createAsyncThunk } from '@reduxjs/toolkit';
+
+const apiUrl = 'https://us-central1-bookstore-api-e63c8.cloudfunctions.net/bookstoreApi/apps/2kecdTt0T9jUNqwudIao/books';
+// Action
+
+const initialState = {
+  isLoading: false,
+  list: null,
+  isError: false,
+};
+
+export const LIST_DATA = createAsyncThunk('LIST_DATA', async () => {
+  const response = await fetch(apiUrl);
+  return response.json();
+});
+export const ADD_BOOK = createAsyncThunk('ADD_BOOK', async (payload) => {
+  await fetch(apiUrl, {
+    method: 'POST',
+    headers: {
+      'content-type': 'application/json',
+    },
+    body: JSON.stringify(payload),
+  });
+  window.location.reload();
+});
+
+export const REMOVE_BOOK = createAsyncThunk('REMOVE_BOOK', (payload) => {
+  fetch(`${apiUrl}/${payload}`, {
+    method: 'DELETE',
+  });
+});
```

Comment:
- [ ] Good job adding using the `createAsyncThunk` in the project for creating thunk action creators. But for the `ADD_BOOK` action though you return a promise you should also automatically dispatch an action type that will pass the book list to the reducer and update the state. currently, I have to reload the page to see the changes. Also, notice its not a good practice to use `window.location.reload();`
- [ ] Also for the `REMOVE_BOOK` you should return new data that updates our state. currently, I have to reload the page to see the changes. This will make your app SPA and Professional.

Reply:
@Tadesse-Alemayehu Tadesse-Alemayehu, two points were raised:-
-  As I can see I am able to add the book and see the refreshed list without reloading the page, please click the live link https://r-ahmed2022.github.io/bookstore/ 
Although,  the window.reload() method may not be the optimized way to do it. If u have a better way, please advise.
-  In my opinion, the only bug in the PR is the list is not automatically updated once I remove the book, and I will fix it, if you a better suggestion , please advise

Reply:
- Dear @r-ahmed2022, Sorry for the late replay, You mentioned `add/remove the book and see the refreshed list without manually reloading the page` But please remember the whole point of using the Redux store is to maintain the state so that in a feature we will be able to update part of your DOM without needing to reload. 

- For the suggestion, You should be able to dispatch an action once the data is ready to be added or removed. 
> I'm only trying to help you understand what you should be doing, And I'm sure you will manage. good luck 🙌